src/common/str_utils.cpp

```diff
@@ -261,6 +261,23 @@ void StringBuilder::append_string(const char *str) {
     *current_pos_ = '\0';
 }
 
+void StringBuilder::append_string_view(string_view_utf8 str) {
+
+    while (true) {
+        if (is_problem()) {
+            return;
+        }
+
+        char b;
+        str.copyToRAM(&b, 1);
```

Comment:
Uff, haven't noticed this before. copyToRam is quite heavy for this, I'd suggest making getbyte public instead.

Reply:
it's now lightweighter :)